### PR TITLE
Use `sphinx_gallery` instead of markdown for examples

### PR DESCRIPTION
**Pull request by bnaul**
Should make using `doctr` easier since everything will be executed by `sphinx` instead of via a separate external pipeline (see https://github.com/cesium-ml/cesium/pull/216).

### CODE REVIEW

**Comment by acrellin** [comment]
Looks good. Works for me on my machine. :+1:

**Comment by bnaul** [comment]
Gonna go ahead and merge so that we can be sure it builds ok on Travis...

**Comment by bnaul** [comment]
@stefanv if you want any style or other changes I'll make a new PR.

File: doc/index.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Machine Learning Time-Series Platform Documentation
 
    Installation <install>
    Time-Series Features <feature_table>
-   Examples <examples>
+   Examples <auto_examples/index>
```
Comment (stefanv) [inline doc/index.rst]: I think the path we used to have was better?

File: examples/plot_EEG_Example.py [hunk 2]
```diff
@@ -71,26 +78,26 @@
...
-[dask graph](http://dask.pydata.org/en/latest/custom-graphs.html). Functions should take
-three arrays `times, measurements, errors` as inputs; details can be found in the
-`cesium.featurize`
-[documentation](https://cesium.readthedocs.org/en/latest/api/cesium.featurize.html).
-Here we'll compute five standard features for EEG analysis provided by [Guo et al.
-(2012)](http://linkinghub.elsevier.com/retrieve/pii/S0957417411003253):
-
-```python
+
+###############################################################################
```
Comment (stefanv) [inline examples/plot_EEG_Example.py]: Do we need to use so many hashes?

File: doc/index.rst (hunk 1 above)
Comment (bnaul) [inline doc/index.rst]: How about `example_src` and `examples`?

**Pull request merged**